apps/dapp/src/providers/SwapProvider.tsx

```diff
@@ -128,18 +101,31 @@ export const SwapProvider = (props: PropsWithChildren<{}>) => {
       });
       return;
     }
-
     setError(null);
 
-    const tokenInfo = getTokenInfo(token);
+    // Initialize vars
+    let receipt: TransactionReceipt | undefined;
+    const swapType: SwapType = SwapType.SwapExactIn;
+    const tokenInfo = getTokenInfo(tokenIn);
     const tokenContract = new ERC20__factory(signer).attach(tokenInfo.address);
     const balance = await tokenContract.balanceOf(wallet);
-    const verifiedAmountIn = amountIn.lt(balance) ? amountIn : balance;
+    const amountIn = quote.swapAmount.lt(balance) ? quote.swapAmount : balance;
```

Comment:
I don't think that this should be changed? It will mean:
  - The user would spend gas which to allow `min(balance, quote.swapAmount)`, but then the transaction will likely fail anyway since the minAmount expected won't be hit (which is part of the quote and left unchanged)

Probably better that it's left as just `amountIn = quote.swapAmount` and then the ensureAllowance will fail?